cleanenv.go

```diff
@@ -198,6 +199,32 @@ func (sm *structMeta) isFieldValueZero() bool {
 	return isZero(sm.fieldValue)
 }
 
+// Any specific supported struct can be added here
+var validStructs = map[string]func(*reflect.Value, string, *string) error{
+	"time.Time": func(field *reflect.Value, value string, layout *string) error {
+		var l string
+		if layout != nil {
+			l = *layout
+		} else {
+			l = time.RFC3339
+		}
+		val, err := time.Parse(l, value)
+		if err != nil {
+			return err
+		}
+		field.Set(reflect.ValueOf(val))
+		return nil
+	},
+	"net/url.URL": func(field *reflect.Value, value string, _ *string) error {
+		val, err := url.Parse(value)
+		if err != nil {
+			return err
+		}
+		field.Set(reflect.ValueOf(*val))
+		return nil
+	},
+}
```

Comment:
Nice one. But I would like to use types instead of strings, so it will work a bit safer.

Also, it will be nice to define a value function as a type.
```suggestion
// parseFunc custom value parser function
type parseFunc func(*reflect.Value, string, *string) error

// Any specific supported struct can be added here
var validStructs = map[reflect.Type]parseFunc{
	reflect.TypeOf(time.Time{}): func(field *reflect.Value, value string, layout *string) error {
		var l string
		if layout != nil {
			l = *layout
		} else {
			l = time.RFC3339
		}
		val, err := time.Parse(l, value)
		if err != nil {
			return err
		}
		field.Set(reflect.ValueOf(val))
		return nil
	},
	reflect.TypeOf(url.URL{}): func(field *reflect.Value, value string, _ *string) error {
		val, err := url.Parse(value)
		if err != nil {
			return err
		}
		field.Set(reflect.ValueOf(*val))
		return nil
	},
}
```